### PR TITLE
NATIVE-291: Missing icon in CategoryList

### PR DESCRIPTION
**Pull request by SimonVortkamp**

### CODE REVIEW

File: src/modules/categories/components/CategoryList.js [hunk 1]
```diff
@@ -18,17 +20,27 @@ type PropsType = {|
   query?: string,
   theme: ThemeType,
   onItemPress: (tile: { title: string, thumbnail: string, path: string }) => void,
-  language: string
+  language: string,
+  thumbnail: string
```
Comment (klinzo) [inline src/modules/categories/components/CategoryList.js]: You should use `thumbnail?: string`, so you don't have to pass anything if no thumbnail should be rendered.

File: src/modules/categories/components/CategoryList.js [hunk 2]
```diff
@@ -18,17 +20,27 @@ type PropsType = {|
...
 /**
  * Displays a ContentList which is a list of categories, a caption and a thumbnail
  */
 class CategoryList extends React.Component<PropsType> {
   render () {
-    const { categories, title, content, query, theme, onItemPress, language } = this.props
+    const { categories, title, content, query, theme, onItemPress, language, thumbnail } = this.props
     return <>
-      {title && <Caption title={title} theme={theme} />}
+      <CategoryThumbnail source={thumbnail} />
```
Comment (klinzo) [inline src/modules/categories/components/CategoryList.js]: Use the suggestion above and `{ thumbnail && <CategoryThumbnail ... /> }` here

File: src/modules/common/components/CategoryListCaption.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+// @flow
+
+import React from 'react'
+import styled, { type StyledComponent } from 'styled-components/native'
+import type { ThemeType } from '../../theme/constants/theme'
+
+const H1: StyledComponent<{}, ThemeType, *> = styled.Text`
```
Comment (klinzo) [inline src/modules/common/components/CategoryListCaption.js]: Better to use meaningful names here, e.g. `CaptionWithThumbnail`

File: src/modules/categories/components/CategoryList.js (hunk 1 above)
Comment (klinzo) [reply src/modules/categories/components/CategoryList.js]: Or `thumbnail: ?string`

File: src/routes/search/components/SearchModal.js [hunk 4]
```diff
@@ -88,7 +88,7 @@ class SearchModal extends React.Component<PropsType, StateType> {
     return <ScrollView contentContainerStyle={{ flexGrow: 1 }}>
       <SpaceBetween>
         <View><CategoryList categories={filteredCategories} query={query} onItemPress={this.onItemPress}
-                            theme={theme} language={language} /></View>
+                            theme={theme} language={language} thumbnail={''} /></View>
```
Comment (Schedulaar) [inline src/routes/search/components/SearchModal.js]:
```suggestion
                            theme={theme} language={language} /></View>
```
Would be cooler, if we could just omit `thumbnail` if it's not necessary. (Can be done using @klinzo's suggestions)

File: src/modules/categories/components/Categories.js [hunk 5]
```diff
@@ -163,7 +163,8 @@ class Categories extends React.Component<PropsType> {
       </SpaceBetween>
     }
     // some level between, we want to display a list
-    return <SpaceBetween>
+    return (
+      <>
```
Comment (Schedulaar) [inline src/modules/categories/components/Categories.js]: The SpaceBetween is needed to give `View` (the actual content) and `SiteHelfulBox` (the feedback prompt) appropriate spacing. Reintroduce it here.

File: src/modules/categories/components/CategoryList.js [hunk 6]
```diff
@@ -18,17 +20,27 @@ type PropsType = {|
...
  * Displays a ContentList which is a list of categories, a caption and a thumbnail
  */
 class CategoryList extends React.Component<PropsType> {
   render () {
-    const { categories, title, content, query, theme, onItemPress, language } = this.props
+    const { categories, title, content, query, theme, onItemPress, language, thumbnail } = this.props
     return <>
-      {title && <Caption title={title} theme={theme} />}
+      {thumbnail && <CategoryThumbnail source={thumbnail} />}
+      {title && <CategoryListCaption title={title} theme={theme} withThumbnail={Boolean(thumbnail)} />}
```
Comment (Schedulaar) [inline src/modules/categories/components/CategoryList.js]: We usually use `!!thumbnail` as a shorthand for `Boolean(thumbnail)`. (you don't have to adjust this.)
```suggestion
      {title && <CategoryListCaption title={title} theme={theme} withThumbnail={!!thumbnail} />}
```

File: src/modules/common/components/CategoryListCaption.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  color: ${props => props.theme.colors.textColor};
+  font-family: ${props => props.theme.fonts.decorativeFontBold};
+`
+
+const CaptionWithThumbnail: StyledComponent<{}, ThemeType, *> = styled.Text`
+  padding: 0 0;
+  font-size: 20px;
+  text-align: center;
+  color: ${props => props.theme.colors.textColor};
+  font-family: ${props => props.theme.fonts.decorativeFontBold};
```
Comment (Schedulaar) [inline src/modules/common/components/CategoryListCaption.js]: You could add `withThumbnail` to the PropsType of the styled component. Then you wouldn't have to define the quasi-same styled-component twice and instead use `padding: ${props => props.withThumbnail ? '0 0' : '20px 0'}`.

**Pull request merged**